protocol/codec.go

```diff
@@ -26,6 +26,9 @@ import (
 	"github.com/algorand/msgp/msgp"
 )
 
+// ErrorInvalidObject is used to state that an object decoding has failed because it's invalid.
+var ErrorInvalidObject = fmt.Errorf("unmarshalled object is invalid")
```

Comment:
Errors are typically start with "Err" and not "Error".

Reply:
Also, since you're not really formatting anything in here, it should be "errors.New(...)"